subprojects/plugins/src/main/java/org/gradle/api/internal/tasks/DefaultSourceSetOutput.java

```diff
@@ -85,6 +91,11 @@ public void addClassesDir(Callable<File> classesDir) {
         classesDirs.from(classesDir);
     }
 
+    @Override
+    public ConfigurableFileCollection getAnnotationProcessorGeneratedSourcesDirs() {
```

Comment:
I think "getGeneratedSourcesDirs" would be better, so we can expand this to other source generators.